### PR TITLE
Simplifying client import to be done once

### PR DESCRIPTION
**Pull request by vsoch**
This pull request will (hopefully) address #200. The core issue was that I had a bad design that looked like this:

 - instantiate client based on environment variables (e.g., is there a database? a client name?)
 - create the parser for the user based on that client
 - based on the command (pull, push) import a main function to run it
 - in the main function import the client again, taking into account the image uri (e.g., s3:// should re-import and use Amazon S3 client)

What seems to be happening (and likely due to the messiness of the above) is that it was switching between 3 states of client, and it for some reason never got to s3. I've been unhappy with this design since I created it, so I thought now would be a good time to fix it up. The fixed design looks like this:

 - create the parser for the user with all possible commands
 - based on the command (pull, push) import a main function to run it
 - in the main function instantiate the client for the first (and final) time, taking into account environment and an image URI simultaneously. If the client doesn't have the function implemented, just tell the user and exit.

The main change to fix this was, instead of instantiating the client in sregistry.main, I removed that entirely and the functions import "get_client" and then use it when appropriate.

Additionally, I went through and updated many "bot.error" messages followed by "sys.exit" to just be "bot.exit" and updated docs.

### CODE REVIEW

**Comment by vsoch** [comment]
@tschoonj please review at your leisure!

**Comment by tschoonj** [comment]
Thanks @vsoch. That will be for tomorrow...

File: docs/getting-started/README.md [hunk 1]
```diff
@@ -55,8 +55,7 @@ The following commands are considered "local" in that they come with every clien
  - *images*: `[local]`: list images in your local database, optionally with filters to search.
  - *inspect* `[local]`: prints out an image manifest and metadata retrieved from its endpoint.
  - *mv*: `[local]`: move a container in your storage to another location on your system, and update the database.
- - *rm* `[local]`: is akin to Docker's remove, and says "remove this record from my database, but don't delete the image." This corresponds with deleting the database record, but not the image file in your storage.
- - *rmi* `[local]`: the same as `rm`, but additionally deletes the image file from storage.
+ - *rm* `[local]`: remove an image from the database.
```
Comment (tschoonj) [inline docs/getting-started/README.md]: Perhaps use the same description as in `commands.md`?:

Replace: `remove an image from the database.` with `remove an image and record from the database.`

File: docs/getting-started/developers.md [hunk 2]
```diff
@@ -105,8 +106,7 @@ We've reviewed these commands before, and will do it again. The following comman
  - *images*: `[local]`: list images in your local database, optionally with filters to search.
  - *inspect* `[local]`: prints out an image manifest and metadata retrieved from its endpoint.
  - *mv*: `[local]`: move a container in your storage to another location on your system, and update the database.
- - *rm* `[local]`: is akin to Docker's remove, and says "remove this record from my database, but don't delete the image." This corresponds with deleting the database record, but not the image file in your storage.
- - *rmi* `[local]`: the same as `rm`, but additionally deletes the image file from storage.
+ - *rm* `[local]`: remove an image.
```
Comment (tschoonj) [inline docs/getting-started/developers.md]: Same here?

**Comment by vsoch** [comment]
This will now close #206 - 

 - container is replaced with image
 - shell is fixed
 - delete / rm are clarified

File: docs/getting-started/developers.md (hunk 2 above)
Comment (vsoch) [reply docs/getting-started/developers.md]: You got it! (for both) :)

**Comment by vsoch** [comment]
@tschoonj I'm going to keep the credentials required for now. We can follow up with another PR.

**Pull request merged**